scala-generator/src/main/scala/models/Play2Json.scala

```diff
@@ -333,9 +333,13 @@ case class Play2Json(
       }
       case fields => {
         Seq(
-          "(",
-          serializations.mkString(" and\n").indent(2),
-          s")(${model.name}.apply _)"
+          "for {",
+          (fields zip serializations)
+            .map { case (field, reader) =>
+              s"${field.name} <- $reader".indent(2)
+            }
+            .mkString("\n"),
+          s"} yield ${model.name}(${fields.map(_.name).mkString(", ")})"
         ).mkString("\n")
```

Comment:
We could get rid of the `Seq().mkString("\n")` thing and do
```scala
val constructorCall = s"${model.name}(${fields.map(_.name).mkString(", ")})"

(fields zip serializations)
  .map { case (field, reader) =>
    s"${field.name} <- $reader".indent(2)
  }
  .mkString("for {", "\n", s"} yield $constructorCall")
```

or, probably even cleaner
```scala
val constructorCall = s"${model.name}(${fields.map(_.name).mkString(", ")})"

val forComprehensions = (fields zip serializations)
  .map { case (field, reader) =>
    s"${field.name} <- $reader".indent(2)
  }

s"""for {
      |${forComprehensions.mkString("\n")}
      } yield $constructorCall""".stripMargin
```

Reply:
agreed - nicer

Reply:
👍